oauth2.go

```diff
@@ -149,6 +151,22 @@ func (c *Config) PasswordCredentialsToken(ctx context.Context, username, passwor
 	})
 }
 
+func (c *Config) getParams(base url.Values) url.Values {
+	var params url.Values
+	if c.Params != nil {
+		params = *c.Params
```

Comment:
This doesn't make a copy of `c.Params`, you have to copy each element in a map as the underlying data structure of a map contains a pointer not the actual data.